docs/how_to_guides.md

```diff
@@ -0,0 +1,87 @@
+
+## How to set up web search with the Toolkit
+
+To use Coral with web search, simply use the `Tavily_Internet_Search` tool by adding your API key to the env file. Alternatively you can use any search provider of your choosing, either with your own implementation or an integration implementation (such as LangChain) by following these [steps below](/docs/custom_tool_guides/tool_guide.md).
+
+## How to set up PDF Upload with the Toolkit
+
+To use Coral with document upload, simply use the `File_Upload_LlamaIndex` or `File_Upload_Langchain` (this needs a cohere API key in the .env file) tool or by adding your API key to the env file. Alternatively you can use any document uploader of your choosing, either with your own implementation or an integration implementation (such as LangChain) by following these [steps below](/docs/custom_tool_guides/tool_guide.md).
+
+## How to create your own tools and retrieval sources
+
+Toolkit includes some sample tools that you can copy to configure your own data sources:
+
+- File loaders - Parses a PDF file and performs RAG. Enables users to upload PDF in Toolkit UI. Users have an option to use either Langchain or Llamaindex, whichever is preferred. Langchain is used by default.
+- Data loaders - This tool queries a data source and then performs RAG on extracted documents. We used Langchain's Wikiretriever as the sample data source.
+- Functions - Python interpreter and calculator tools.
+
+To create your own tools or add custom data sources, see our guide: [tools and retrieval sources overview](/docs/custom_tool_guides/tool_guide.md)
+
+# Use Experimental Features
+
+**Please note that these are experimental features.**
+
+## LangChain MultiHop
+
+Chatting with multi hop tool usage through LangChain is enabled by setting experimental feature flag to True in `.env`. 
+
+```bash
+USE_EXPERIMENTAL_LANGCHAIN=True
+```
+
+By setting this flag to true, only tools that have a LangChain implementation can be utilized. 
+These exist under `LANGCHAIN_TOOLS` and require a `to_lanchain_tool()` function on the tool implementation which returns a LangChain compatible tool. 
+Python interpreter and Tavily Internet search are provided in the toolkit by default once the environment is set up.
+
+Example API call:
+```bash
+curl --location 'http://localhost:8000/langchain-chat' \
+--header 'User-Id: me' \
+--header 'Content-Type: application/json' \
+--data '{
+    "message": "Tell me about the aya model",
+    "tools": [{"name": "Python_Interpreter"},{"name": "Internet Search"},]
+}'
+```
+
+Currently, citations are not supported in LangChain multi hop.
```

Comment:
Can we have a `how to use community features` section here? It only requires setting `use_community_feature=True` by either running `make first-run`, `make-setup` or manually adding it to the .env file

Reply:
[Is section sufficient? ](https://github.com/cohere-ai/cohere-toolkit/pull/107/files#diff-ed3c6a99654b2c18794f310672d9cfa21645f6e0df3ea53a4bd1d425952eb983R4)